### ISSUE TITLE
dmonsfree: 1 removed doesn't match 0 pending on Fort Ludios 

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#371]
```
Suddenly, the dungeon collapses.
dmonsfree: 1 removed doesn't match 0 pending on Fort Ludios 
Generating more information you may report:

#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=139633631864768) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=139633631864768) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=139633631864768, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007efefd1c0476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007efefd1a67f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x000055c76ca1c1c0 in NH_abort () at end.c:236
#6  0x000055c76ca23239 in panic (str=0x55c76d5a0140 "%s") at end.c:802
#7  0x000055c76cf9cd2e in impossible (s=0x55c76d553860 "dmonsfree: %d removed doesn't match %d pending on %s") at pline.c:685
#8  0x000055c76cda28ce in dmonsfree () at mon.c:3106
#9  0x000055c76cd8dff5 in movemon () at mon.c:1585
#10 0x000055c76c71c22c in moveloop (resuming=0 '\000') at allmain.c:247
#11 0x000055c76d4236c8 in main (argc=0, argv=0x7ffcada36838) at ../sys/unix/unixmain.c:353


```

**Comment by elunna** [comment]
Extra info
```
(rr) p toplines
$1 = "  (0 pts.)  (0 pts.)\000 high security area!  An alarm sounds!", '\000' <repeats 240 times>

(rr) p moves
$2 = 1434844

Move 1433923: A PM_GIANT_LOUSE was xkilled...
Move: 1434815: A PM_ARCTIC_FERN_SPORE was mondead...
```

**Comment by elunna** [comment]
This could be related (https://github.com/NetHack/NetHack/commit/263bd05fa0e0029eaa9d1e12ab49fd6d4e83faeb), however that commit has been patched already. But the concept is that a player poly'd into a vampire drains a monster to 0 HP but didn't outright kill it, triggering the "1 removed doesn't match 0 pending".

**Comment by elunna** [comment]
Also got a dmonsfree: 2 removed doesn't match 1 pending on Home 1 
For Tourist

**Comment by elunna** [comment]
This might be fixed with the recent drain life fixes in 895bb39f6
